metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java

```diff
@@ -46,35 +54,113 @@
     public static final String DC_IDENTIFIER = "DC.identifier";
     public static final String DC_LANGUAGE = "DC.language";
     public static final String CITATION_TITLE = "citation_title";
+    public static final String DATE_SEPARATOR = "-";
+    public static final String ARTICLE_HTML = "article.html";
+    public static final String YEAR_ONLY = "2001";
+    public static final String FULL_DATE = "2001-12-19";
+    public static final String DC_DATE = "dc.date";
+    public static final String VALID_DATE = "2017";
 
     private WireMockServer wireMockServer;
 
+    @AfterEach
+    void tearDown() {
+        wireMockServer.stop();
+    }
+
     @ParameterizedTest(name = "#{index} - {0}")
     @MethodSource({
         "provideMetadataWithLowercasePrefixes",
         "provideMetadataForTags",
         "provideMetadataForAbstract",
         "provideMetadataForTitle",
-        "provideMetadataForDate",
         "provideMetadataForContributors",
         "provideMetadataForIdentifier",
         "provideMetadataForLanguage"
     })
     public void getCreatePublicationParsesHtmlAndReturnsMetadata(String testDescription, String html,
                                                                  CreatePublicationRequest expectedRequest)
         throws IOException {
-        String filename = "article.html";
-        URI uri = prepareWebServerAndReturnUriToMetadata(filename, html);
+        URI uri = prepareWebServerAndReturnUriToMetadata(ARTICLE_HTML, html);
         MetadataService metadataService = new MetadataService();
         Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
 
-        assertThat(request.isPresent(), is(true));
-        CreatePublicationRequest actual = request.get();
+        CreatePublicationRequest actual = request.orElseThrow();
         actual.setContext(null);
 
         assertThat(actual, is(equalTo(expectedRequest)));
+    }
 
-        wireMockServer.stop();
+    @ParameterizedTest(name = "getCreatePublication returns date when date attribute {0} with value {1}")
+    @ArgumentsSource(DateArgumentsProvider.class)
+    void getCreatePublicationReturnsDateWhenDateVariantIsPresent(String tagAttribute, String date) throws IOException {
+        String html = generateHtmlWithSingleMetaTag(tagAttribute, date);
+        URI uri = prepareWebServerAndReturnUriToMetadata(ARTICLE_HTML, html);
+        MetadataService metadataService = new MetadataService();
+        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
+
+        CreatePublicationRequest actual = request.orElseThrow();
+        actual.setContext(null);
+
+        CreatePublicationRequest expectedRequest = getCreatePublicationRequestWithDateOnly(date);
+        assertThat(actual, is(equalTo(expectedRequest)));
+    }
+
+    @Test
```

Comment:
Now we're mixing @Test with descriptive method name

Reply:
@mattiahj  I am not sure I understand your comment